Fix build with newer CMake and LLVM versions.

cassandras-lies opened the pull request:
This is needed to build on Mac now.

Orphis committed:
Initial commit

Orphis committed:
Standalone build detection

A standalone build will have all targets added to “all”
and allow building tests later.

Orphis committed:
Detect invalid BOOST_SOURCE folders

Orphis committed:
Boost version detection

Orphis committed:
Add README and LICENSE

Orphis committed:
Add Boost Exception library

Orphis committed:
Add Boost Graph library

Orphis committed:
Add Boost Serialization library

Orphis committed:
Fix error detection in download logic

Orphis committed:
Fix Boost Test

Orphis committed:
Change library output name to make Boost release

Orphis committed:
Upgrade to Boost 1.63

Orphis committed:
Comment missing modules

Orphis committed:
Add signals library

Orphis committed:
Add type_erasure library

Orphis committed:
Add wave library

Orphis committed:
Add math library

Orphis committed:
Add complex library

Untested on Windows yet

Orphis committed:
Add coroutine library

Orphis committed:
Add fiber library

Orphis committed:
Add mph and graph_parallel

Orphis committed:
Add TravisCI support

Orphis committed:
Add TravisCI build status tag to README.md

Orphis committed:
Add iOS support to Travis

Orphis committed:
Add support for ccache in standalone builds

Orphis committed:
Add function for wrapping tests

Orphis committed:
Add missing dependency on system in filesystem

Orphis committed:
Add system tests

Orphis committed:
Add missing file in UTF

Orphis committed:
Add missing dependency to thread

Orphis committed:
Fix context library

Orphis committed:
Add atomic tests

Orphis committed:
Add container tests

Orphis committed:
Add coroutine tests

Orphis committed:
Add some date_time tests

Orphis committed:
When building standalone, don’t setup any configuration

Orphis committed:
Add context tests

Orphis committed:
Target iOS 9.0 for the fiber library

Orphis committed:
Add exception tests

Orphis committed:
Add program_options test

Orphis committed:
Add missing dependency on system in random

Orphis committed:
Add fiber tests

Orphis committed:
Fix Android ASM compilation

Orphis committed:
Don't link chrono against RT on Android

Orphis committed:
Fix CheckPreprocessor cache

Zer0xFF committed:
Update CMAKE_MODULE_PATH
this should allow the script to find the correct module path even when being called externally

Orphis committed:
Merge pull request #10 from Thunder07/patch-2

Update CMAKE_MODULE_PATH

Thunder07 committed:
context.cmake fix typo

Orphis committed:
Merge pull request #11 from Thunder07/patch-3

Fix context on Windows

Orphis committed:
Ignore all build* folders

Orphis committed:
Save downloaded archive to a file with same name

Orphis committed:
Add script to repack Boost, patch it and remove extra files

Extra files are for example documentation files to slim down the archive to download and the number of files a lot.
Down from 71MB to 22MB.

Orphis committed:
Add filesystem tests

Orphis committed:
Add random tests

Orphis committed:
Point default download to repackaged Boost

Fixes some Boost regressions on macOS and Android

Orphis committed:
Change ignore for downloaded archive to match tar.xz

Orphis committed:
Add more missing dependencies

Orphis committed:
Switch TravisCI to Xcode 8.2

Orphis committed:
Add Appveyor CI

Orphis committed:
Set /bigobj on tests for Windows

Orphis committed:
Use GCC 5 on Linux

Orphis committed:
Merge pull request #14 from Orphis/tests

Add tests!

Orphis committed:
Fix message() call to output to stdout

Orphis committed:
Properly expose/hide variables in cache

Orphis committed:
Add locale library

Orphis committed:
Bump CMake to 3.7.2 in Travis / Linux

Orphis committed:
Update to NDK r13b

Orphis committed:
Fix ccache

Orphis committed:
Split top level build script in many files

Orphis committed:
Add tests and support for zlib and bzip2 in iostreams

Orphis committed:
Switch hardcoded -pthread to find_package(Threads) instead

Orphis committed:
Add FreeBSD support in context

Orphis committed:
Add Boost Log library

Orphis committed:
Use check_cxx_symbol_exists in check_preprocessor

tijme committed:
Fixed invalid reference to parallel source file.

Orphis committed:
android: Upgrade to NDK r14b

Orphis committed:
Put Boost libraries and tests into IDE folders

Orphis committed:
Add documentation for fetching an alternate Boost archive

Orphis committed:
Disable failing test in Boost locale with 1.64 on macOS

Orphis committed:
Update to Boost 1.64.0

Orphis committed:
Point repack script to 1.64.0

ekmett committed:
Use /W0

Using /W0 rather than /w means that when the Platform cmake sets up a build with /W3, you don't get unignorable D9025 warnings on windows. The options otherwise do the same thing.

Orphis committed:
Build with Visual Studio 2015 and 2017

rcdailey committed:
Allow using a local directory to find boost archives

Instead of downloading boost, allow the user to specify the path to a directory
that contains the required boost archive. To specify this archive directory, use
the new BOOST_ARCHIVE_DIRECTORY cache option. It must point to a valid, existing
directory on the filesystem. To instead download the archives normally, simply
set BOOST_ARCHIVE_DIRECTORY to an empty string or delete it entirely.

When a local directory for archives is being used, any SHA mismatch or missing
archive file will result in permanent failure. In other words, boost will not be
downloaded if these checks fail.

helmesjo committed:
Split up BOOST_LIBS into _OPTIONAL & _REQUIRED lists (currently only 'header' as required). Only sets up target for libs found in BOOST_SOURCE/libs/libname.
This allows to build boost with only a subset of the libs (if for example using bcp).

Fixed copy-paste typo

Renamed cmake-files in /libs to equal corresponding module-name in boost-repo/libs.

Moved 'added <lib>'-message into the script actually adding the lib.

Removed 'added lib'-message

Commit:
Updated to Boost 1.67.0

Orphis committed:
Use repacked version of Boost 1.67.0

rcdailey committed:
CMake: Add cache option to disable building of test targets (#55)

The `BOOST_DISABLE_TESTS` option (disabled by default) controls whether or
not the test targets should be generated by CMake. Enabling this is useful
in cases where you just want to build boost and install it, without incurring
the additional overhead (build time) of building the tests.

janbar committed:
first commit

janbar committed:
cmake project for openssl-1.1.0

Orphis committed:
Fix embarrassing typo

janbar committed:
1.1.1

janbar committed:
cmake project for openssl-1.1.1

Orphis committed:
Set up CI with Azure Pipelines

Orphis committed:
Bump CMake requirement to 3.12 (#62)

Orphis committed:
Upgrade to 1.69.0 (#67)

gegles committed:
Fix issue with -isystem when compiling swift code. (#72)

gegles committed:
Allow BOOST_LIBS_OPTIONAL to be overwritten by user. (#73)

Orphis committed:
Change some set_target_properties to target_compile_definitions() (#74)

Orphis committed:
Use FetchContent to download Boost sources (#68)

FetchContent can be overriden in the parent project with
a lot more options. Should be more robust overall than my
custom download method.

WenbinHou committed:
Corrent FETCHCONTENT_SOURCE_DIR_boost 

It should be FETCHCONTENT_SOURCE_DIR_BOOST.
The upper-cased **BOOST**, according to CMake FetchContent documentation.

janbar committed:
Merge pull request #2 from janbar/openssl-1.1.1

Openssl 1.1.1

janbar committed:
fix build for arm platform

Orphis committed:
Upgrade to Boost 1.70.0

awakecoding committed:
add support for MSVC compiler

janbar committed:
fix import of the project

janbar committed:
1.1.1b

janbar committed:
cmake project for openssl-1.1.1b

janbar committed:
Merge pull request #4 from janbar/openssl-1.1.1b

Openssl 1.1.1b

viettrungluu committed:
Print a message before/after fetching Boost

It can take a while to fetch Boost, so it's nice to be told what it's doing (otherwise you might think that CMake is hung).

(Arguably, this should be a part of FetchContent_Populate.... *shrug*)

gegles committed:
Bump default to Boost 1.71

janbar committed:
android: adding openssl support for qt-5.12

janbar committed:
WINDOWS: fix build

janbar committed:
WINDOWS: rename output

janbar committed:
1.1.1d

janbar committed:
revert part of commit 3251ab8531bb5b0b45e929b20276c4fb8bdd2fba

janbar committed:
configure dso

janbar committed:
fix cpack filename

janbar committed:
Merge pull request #9 from janbar/1.1.1d

1.1.1d

janbar committed:
cleanup cpack config

janbar committed:
target include directories

janbar committed:
fill build info

janbar committed:
add missed definitions

janbar committed:
oops revert a redefinition

janbar committed:
MSVC: don't erase CXX flags with C flags

janbar committed:
1.1.1e

janbar committed:
update project cmake for 1.1.1e

janbar committed:
1.1.1g

Orphis committed:
Update macOS image name

Use macOS latest instead of a specific and old version.

janbar committed:
1.1.1k

janbar committed:
update project cmake for 1.1.1k

charliekang committed:
Update CMakeLists.txt

Bintray sunset - change boost URL

janbar committed:
do not create target uninstall for subproject

raaidrt committed:
Modified boost source

janbar committed:
1.1.1m

janbar committed:
fix apps link on SunOS

Add target link to socket, nsl

Commit:
OpenSSL 1.1.1n

janbar committed:
Merge pull request #27 from ekcdd/master

OpenSSL 1.1.1n

janbar committed:
1.1.1s

reubenyap committed:
Initial commit

levonpetrosyan93 committed:
First commit

janbar committed:
fix #31

janbar committed:
1.1.1t

levonpetrosyan93 committed:
Aligned with main repository

levonpetrosyan93 committed:
Add some spend related apis

levonpetrosyan93 committed:
Adding full api test and more functions

Commit:
Update CMakeLists.txt

levonpetrosyan93 committed:
Sync with main repo

levonpetrosyan93 committed:
Full Spark spend creation implemented

levonpetrosyan93 committed:
Small refactor

janbar committed:
1.1.1w

janbar committed:
fix cmake project 1.1.1w

janbar committed:
Merge pull request #32 from gmh5225/master

Fix 'cmake_minimum_required() should be called prior to this top-level project()'

janbar committed:
configure the package config file openssl.pc

levonpetrosyan93 committed:
Typo fix

levonpetrosyan93 committed:
Sync with main repo

levonpetrosyan93 committed:
Sync with main repo

levonpetrosyan93 committed:
return spent coins

levoncrypto committed:
Script for spark name (#3)

* Ownership proof

* Script for spark name

* Move function declaration

---------

Co-authored-by: levonpetrosyan93 <petrosyan.levon93@gmail.com>

levonpetrosyan93 committed:
Crash on ios fixed

julian-CStack committed:
Fix compilation (#4)

levoncrypto committed:
Test for spark name script (#5)

* Test for spark name script

* Fixes for spark name tests

* Add spark namespace

* Ownership proof verify in spark name test

---------

Co-authored-by: levonpetrosyan93 <45027856+levonpetrosyan93@users.noreply.github.com>

julian-CStack commented:
Current main branch should work fine on recent macos unless Apple has pushed updates in the last week that change that. 
I did notice https://github.com/cassandras-lies/flutter_libsparkmobile/tree/fix-build is 17 commits behind cypherstack:main

cassandras-lies commented:
Yes, it broke for me within the last week. Try `brew update && brew upgrade`. It's actually an upgrade to cmake 4.0.0 in brew and then the Apple update to clang version 17.0.0

```
~/Code/cflsm/scripts/macos $ git show --oneline -1                                                                                                                                                                    
e8c5026 (HEAD -> main, origin/main, origin/HEAD) Merge pull request #40 from cypherstack/win-patch-fix
~/Code/cflsm/scripts/macos $ ./build_all.sh
-- The C compiler identification is AppleClang 17.0.0.17000013
-- The CXX compiler identification is AppleClang 17.0.0.17000013
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: ~/.bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: ~/.bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
~/Code/cflsm/src/deps/boost-cmake ~/Code/cflsm
~/Code/cflsm
CMake Warning (dev) at /opt/homebrew/share/cmake/Modules/FetchContent.cmake:1373 (message):
  The DOWNLOAD_EXTRACT_TIMESTAMP option was not given and policy CMP0135 is
  not set.  The policy's OLD behavior will be used.  When using a URL
  download, the timestamps of extracted files should preferably be that of
  the time of extraction, otherwise code that depends on the extracted
  contents might not be rebuilt if the URL changes.  The OLD behavior
  preserves the timestamps from the archive instead, but this is usually not
  what you want.  Update your project to the NEW behavior or specify the
  DOWNLOAD_EXTRACT_TIMESTAMP option with a value of true to avoid this
  robustness issue.
Call Stack (most recent call first):
  deps/boost-cmake/CMakeLists.txt:10 (FetchContent_Declare)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Fetching Boost
CMake Warning (dev) at /opt/homebrew/share/cmake/Modules/FetchContent.cmake:1953 (message):
  Calling FetchContent_Populate(Boost) is deprecated, call
  FetchContent_MakeAvailable(Boost) instead.  Policy CMP0169 can be set to
  OLD to allow FetchContent_Populate(Boost) to be called directly for now,
  but the ability to call it with declared details will be removed completely
  in a future version.
Call Stack (most recent call first):
  deps/boost-cmake/CMakeLists.txt:19 (FetchContent_Populate)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Fetching Boost - done
-- Removing: ~/Code/cflsm/scripts/macos/build/_deps/boost-src/boost/thread/pthread/thread_data.hpp
-- Copying: ~/Code/cflsm/src/deps/boost-cmake/../thread_data.hpp to ~/Code/cflsm/scripts/macos/build/_deps/boost-src/boost/thread/pthread
-- Boost found: 1.71.0 ~/Code/cflsm/scripts/macos/build/_deps/boost-src
-- Looking for __linux__
-- Looking for __linux__ - not found
-- Looking for _WIN32
-- Looking for _WIN32 - not found
-- Looking for __APPLE__
-- Looking for __APPLE__ - found
-- Looking for __ANDROID__
-- Looking for __ANDROID__ - not found
-- Looking for __FreeBSD__
-- Looking for __FreeBSD__ - not found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE
-- The ASM compiler identification is AppleClang
-- Found assembler: ~/.bin/cc
-- Found BZip2: /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/lib/libbz2.tbd (found version "1.0.8")
-- Looking for BZ2_bzCompressInit
-- Looking for BZ2_bzCompressInit - found
-- Found ZLIB: /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/lib/libz.tbd (found version "1.2.12")
-- Failed to find all ICU components (missing: ICU_LIBRARY uc dt i18n) (found version "76.1")
-- Found iconv library: /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/lib/libiconv.tbd
-- Checking for module 'mpi-c'
--   Package 'mpi-c' not found
-- Checking for module 'mpi-cxx'
--   Package 'mpi-cxx' not found
CMake Error at deps/openssl-cmake/CMakeLists.txt:26 (cmake_minimum_required):
  Compatibility with CMake < 3.5 has been removed from CMake.

  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
  to tell CMake that the project requires at least <min> but has been updated
  to work with policies introduced by <max> or earlier.

  Or, add -DCMAKE_POLICY_VERSION_MINIMUM=3.5 to try configuring anyway.


-- Configuring incomplete, errors occurred!
cp: flutter_libsparkmobile.framework: No such file or directory
```

julian-CStack reviewed (dismissed):
LGTM

Works on updated macOS system. Needs to be double checked on other platforms so I will leave open for now

julian-CStack commented:
@sneurlax if you have the debug output saved still please link/post here.

sneurlax commented:
@cassandras-lies I get this error, revolving error MSB8066: https://gist.github.com/sneurlax/41d54b4f34fcf97def234be66fda1986

julian-CStack commented:
> @cassandras-lies I get this error, revolving error MSB8066: https://gist.github.com/sneurlax/41d54b4f34fcf97def234be66fda1986
> 
> The failure to apply patches is inconsequential, the same error occurs on a fresh clone.

I believe the failure to apply patches earlier in the build process is what you are referring to and those don't appear in that gist.

It looks like the `apply_patches` is failing in cmake https://gist.github.com/sneurlax/41d54b4f34fcf97def234be66fda1986#file-gistfile1-txt-L205-L227

cassandras-lies commented:
I've updated the patch so it will work without errors, including on Windows though I haven't tried it there. Note that running ./scripts/prebuild.{sh,bat} will reset and clean source directories to those specified in .gitmodules; if you'd prefer dirty directories be unpatched, tell me and I'll update it to do that instead.

julian-CStack commented:
> I've updated the patch so it will work without errors, including on Windows though I haven't tried it there. Note that running ./scripts/prebuild.{sh,bat} will reset and clean source directories to those specified in .gitmodules; if you'd prefer dirty directories be unpatched, tell me and I'll update it to do that instead.

This is fine.

Building on windows still fails though:
```
...
[  +15 ms] -- Build files have been written to: C:/Users/julian/Desktop/flutter_libsparkmobile/example/build/windows/x64
[  +14 ms] executing: C:\Program Files\Microsoft Visual Studio\2022\Community\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin\cmake.exe --build C:\Users\julian\Desktop\flutter_libsparkmobile\example\build\windows\x64 --config Debug --target INSTALL --verbose
[  +83 ms] Change Dir: 'C:/Users/julian/Desktop/flutter_libsparkmobile/example/build/windows/x64'
[   +5 ms] Run Build Command(s): "C:/Program Files/Microsoft Visual Studio/2022/Community/MSBuild/Current/Bin/amd64/MSBuild.exe" INSTALL.vcxproj /p:Configuration=Debug /p:Platform=x64 /p:VisualStudioVersion=17.0 /v:n
[  +30 ms] MSBuild version 17.13.19+0d9f5a35a for .NET Framework
[  +17 ms] Build started 2025-05-05 15:36:39.
[  +93 ms] Project "C:\Users\julian\Desktop\flutter_libsparkmobile\example\build\windows\x64\INSTALL.vcxproj" on node 1 (default targets).
[        ] Project "C:\Users\julian\Desktop\flutter_libsparkmobile\example\build\windows\x64\INSTALL.vcxproj" (1) is building "C:\Users\julian\Desktop\flutter_libsparkmobile\example\build\windows\x64\ZERO_CHECK.vcxproj" (2) on node 1 (default targets).
[        ] PrepareForBuild:
[        ]   Creating directory "x64\Debug\ZERO_CHECK\".
[   +2 ms]   Structured output is enabled. The formatting of compiler diagnostics will reflect the error hierarchy. See https://aka.ms/cpp/structured-output for more details.
[        ]   Creating directory "x64\Debug\ZERO_CHECK\ZERO_CHECK.tlog\".
[   +1 ms] InitializeBuildStatus:
[        ]   Creating "x64\Debug\ZERO_CHECK\ZERO_CHECK.tlog\unsuccessfulbuild" because "AlwaysCreate" was specified.
[        ]   Touching "x64\Debug\ZERO_CHECK\ZERO_CHECK.tlog\unsuccessfulbuild".
[  +43 ms] CustomBuild:
[        ]   1>Checking Build System
[  +48 ms] FinalizeBuildStatus:
[        ]   Deleting file "x64\Debug\ZERO_CHECK\ZERO_CHECK.tlog\unsuccessfulbuild".
[        ]   Touching "x64\Debug\ZERO_CHECK\ZERO_CHECK.tlog\ZERO_CHECK.lastbuildstate".
[        ] Done Building Project "C:\Users\julian\Desktop\flutter_libsparkmobile\example\build\windows\x64\ZERO_CHECK.vcxproj" (default targets).
[  +21 ms] Project "C:\Users\julian\Desktop\flutter_libsparkmobile\example\build\windows\x64\INSTALL.vcxproj" (1) is building "C:\Users\julian\Desktop\flutter_libsparkmobile\example\build\windows\x64\ALL_BUILD.vcxproj" (3) on node 1 (default targets).
[        ] Project "C:\Users\julian\Desktop\flutter_libsparkmobile\example\build\windows\x64\ALL_BUILD.vcxproj" (3) is building
"C:\Users\julian\Desktop\flutter_libsparkmobile\example\build\windows\x64\plugins\flutter_libsparkmobile\shared\deps\boost-cmake\apply_patches.vcxproj" (4) on node 1 (default targets).
[        ] PrepareForBuild:
[        ]   Creating directory "x64\Debug\apply_patches\".
[        ]   Structured output is enabled. The formatting of compiler diagnostics will reflect the error hierarchy. See https://aka.ms/cpp/structured-output for more details.
[        ]   Creating directory "x64\Debug\apply_patches\apply_patches.tlog\".
[   +1 ms] InitializeBuildStatus:
[        ]   Creating "x64\Debug\apply_patches\apply_patches.tlog\unsuccessfulbuild" because "AlwaysCreate" was specified.
[        ]   Touching "x64\Debug\apply_patches\apply_patches.tlog\unsuccessfulbuild".
[  +16 ms] CustomBuild:
[        ]   "The build of 'C:\Users\julian\Desktop\flutter_libsparkmobile\example\build\windows\x64\CMakeFiles\72cc91952f6a05d2af8f91fc41a1a8c9\apply_patches.rule' depends on 'C:\USERS\julian\DESKTOP\FLUTTER_LIBSPARKMOBILE\EXAMPLE\BUILD\WINDOWS\X64\_DEPS\BOOST-SRC\.PATCHED'
which is produced by the build of 'C:\Users\julian\Desktop\flutter_libsparkmobile\example\build\windows\x64\CMakeFiles\24e89584e84b787d6ae3e0ff14860c9a\.patched.rule'. The items cannot be built in parallel."
[  +37 ms]   Applying all patches in C:/Users/julian/Desktop/flutter_libsparkmobile/example/windows/flutter/ephemeral/.plugin_symlinks/flutter_libsparkmobile/src/deps/boost-cmake/patches
[  +10 ms]   Applying patch: C:/Users/julian/Desktop/flutter_libsparkmobile/example/windows/flutter/ephemeral/.plugin_symlinks/flutter_libsparkmobile/src/deps/boost-cmake/patches/integral_wrapper.patch
[  +12 ms]   no such file or directory
[   +5 ms] C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.CppCommon.targets(254,5): error MSB8066: Custom build for
'C:\Users\julian\Desktop\flutter_libsparkmobile\example\build\windows\x64\CMakeFiles\24e89584e84b787d6ae3e0ff14860c9a\.patched.rule;C:\Users\julian\Desktop\flutter_libsparkmobile\example\build\windows\x64\CMakeFiles\72cc91952f6a05d2af8f91fc41a1a8c9\apply_patches.rule;C:\Users\e
ric\Desktop\flutter_libsparkmobile\example\windows\flutter\ephemeral\.plugin_symlinks\flutter_libsparkmobile\src\deps\boost-cmake\CMakeLists.txt' exited with code 1.
[C:\Users\julian\Desktop\flutter_libsparkmobile\example\build\windows\x64\plugins\flutter_libsparkmobile\shared\deps\boost-cmake\apply_patches.vcxproj]
[        ] Done Building Project "C:\Users\julian\Desktop\flutter_libsparkmobile\example\build\windows\x64\plugins\flutter_libsparkmobile\shared\deps\boost-cmake\apply_patches.vcxproj" (default targets) -- FAILED.
[        ] Done Building Project "C:\Users\julian\Desktop\flutter_libsparkmobile\example\build\windows\x64\ALL_BUILD.vcxproj" (default targets) -- FAILED.
[        ] Done Building Project "C:\Users\julian\Desktop\flutter_libsparkmobile\example\build\windows\x64\INSTALL.vcxproj" (default targets) -- FAILED.
[   +1 ms] Build FAILED.
[        ] "C:\Users\julian\Desktop\flutter_libsparkmobile\example\build\windows\x64\INSTALL.vcxproj" (default target) (1) ->
[        ] "C:\Users\julian\Desktop\flutter_libsparkmobile\example\build\windows\x64\ALL_BUILD.vcxproj" (default target) (3) ->
[        ] "C:\Users\julian\Desktop\flutter_libsparkmobile\example\build\windows\x64\plugins\flutter_libsparkmobile\shared\deps\boost-cmake\apply_patches.vcxproj" (default target) (4) ->
[        ] (CustomBuild target) ->
[        ]   C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.CppCommon.targets(254,5): error MSB8066: Custom build for
'C:\Users\julian\Desktop\flutter_libsparkmobile\example\build\windows\x64\CMakeFiles\24e89584e84b787d6ae3e0ff14860c9a\.patched.rule;C:\Users\julian\Desktop\flutter_libsparkmobile\example\build\windows\x64\CMakeFiles\72cc91952f6a05d2af8f91fc41a1a8c9\apply_patches.rule;C:\Users\e
ric\Desktop\flutter_libsparkmobile\example\windows\flutter\ephemeral\.plugin_symlinks\flutter_libsparkmobile\src\deps\boost-cmake\CMakeLists.txt' exited with code 1.
[C:\Users\julian\Desktop\flutter_libsparkmobile\example\build\windows\x64\plugins\flutter_libsparkmobile\shared\deps\boost-cmake\apply_patches.vcxproj]
[        ]     0 Warning(s)
[        ]     1 Error(s)
[        ] Time Elapsed 00:00:00.30
[  +12 ms] Building Windows application... (completed in 90.6s)
[        ] Error: Build process failed.
[        ] "flutter run" took 91,197ms.
[   +1 ms]
           #0      throwToolExit (package:flutter_tools/src/base/common.dart:34:3)
           #1      RunCommand.runCommand (package:flutter_tools/src/commands/run.dart:931:9)
           <asynchronous suspension>
           #2      FlutterCommand.run.<anonymous closure> (package:flutter_tools/src/runner/flutter_command.dart:1558:27)
           <asynchronous suspension>
           #3      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:154:19)
           <asynchronous suspension>
           #4      CommandRunner.runCommand (package:args/command_runner.dart:212:13)
           <asynchronous suspension>
           #5      FlutterCommandRunner.runCommand.<anonymous closure> (package:flutter_tools/src/runner/flutter_command_runner.dart:496:9)
           <asynchronous suspension>
           #6      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:154:19)
           <asynchronous suspension>
           #7      FlutterCommandRunner.runCommand (package:flutter_tools/src/runner/flutter_command_runner.dart:431:5)
           <asynchronous suspension>
           #8      run.<anonymous closure>.<anonymous closure> (package:flutter_tools/runner.dart:98:11)
           <asynchronous suspension>
           #9      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:154:19)
           <asynchronous suspension>
           #10     main (package:flutter_tools/executable.dart:99:3)
           <asynchronous suspension>


[        ] Running 3 shutdown hooks
[   +4 ms] Shutdown hooks complete
[   +1 ms] exiting with code 1
```

cassandras-lies commented:
So I had used `patch` in the CMakeLists.txt. I've changed it to copy hardcoded files rather than patch the existing ones. I tested it on Windows this time.

levonpetrosyan93 committed:
Refactore to match spark name changes

julian-CStack committed:
update header to match impl (#6)

julian-CStack commented:
> So I had used `patch` in the CMakeLists.txt. I've changed it to copy hardcoded files rather than patch the existing ones. I tested it on Windows this time.

I would like if you could include the files separately rather than as part of the patch. Any updates or changes in the future will be a bit easier to handle this way. Please include links where the files were sourced from. I regret not doing so for the `thread_data.hpp` file. Feel free to leave the links in a comment here and I will put them along the the source of `thread_data.hpp` somewhere in the repo once I track down where I found it.

cassandras-lies commented:
This will make a submodule dependency dependent on the main repo. Maybe it would be better to vendor it in that case?

The modified files are the result of applying the previous patches to the source tree the boost-cmake module fetches, which is an extracted .bz2 file given as a source from the boost site.

julian-CStack commented:
The boost-cmake submodule only depends on the main repo specifically for use in the main repo. Maybe I'm misunderstanding? 

`src/deps/patches/boost-patch.patch` would still need to exist but I was referring to the new files created in the patch which are not copied from the boost source archive or modified from what I am seeing.

sidenote:  This change https://github.com/cypherstack/flutter_libsparkmobile/pull/41/files#diff-f9b13aa9e581ca347c8e7e7e2bfb45ffa09d53074a461c38e90d4c33886e532bR36 is overridden by https://github.com/cypherstack/flutter_libsparkmobile/blob/main/src/CMakeLists.txt#L46-L47

julian-CStack commented:
I'd actually love to not have to patch anything in the submodules. Not sure if that is feasible thoguh

cassandras-lies commented:
So basically this boost-cmake submodule is pulling boost from a remote server (it does check the checksum), and then building it on the local machine with some CMake files that they've added. Unfortunately the remote boost version used there does not properly support compilation with newer LLVM and CMake. The boost-cmake.patch file adds a new commit in the boost-cmake repository, and that commit contains the updated files. It's kind of a confusing structure. :) Basically, patching the files directly from the outer module would leave a dirty working directory in the submodule (which will cause warnings every time you run `git status`), so the patch files to the boost source live inside the new boost-cmake.patch commit, and then those are copied over. I guess we can have the boost-cmake.patch reference files in the outer directory since it's a patch that exists only in our repo and not on any actual outside server, but I found that aesthetically displeasing.

As to not patching anything, unfortunately the boost-cmake source needs to be patched to work. boost-cmake uses a very old version of boost which is not compatible with new compilers.

Personally I would suggest removing the submodule entirely and vendoring the boost-cmake directory, as it looks like it's a dead project anyway.

Also, with regard to patching submodules generally, they will cause you to reference non-existent commits which will be annoying to work with when you make new commits as you need to be sure not to commit references to the now patched modules (as they can't be pulled from the remote repos specified in .gitmodules).

julian-CStack commented:
Messy. Vendoring it may be the way to go then? Might be a bit of a pain to set stuff up for easy compilation across platforms (looking at you iOS)

Maybe we can user a recent version of boost? (just update the boost download link and hash in https://github.com/cypherstack/flutter_libsparkmobile/blob/main/src/CMakeLists.txt#L46-L47). Not sure if that will be compatible with the current source of sparkmobile. I thought firo uses 1.7x but I could be wrong there.

cassandras-lies commented:
The whole "patching submodules" thing is kind of iffy entirely, but there was already a windows patch that did it, so I did it too. Honestly you should probably fork the submodule and point it to your own patched version rather than keeping a patch in an outer repository, due to the dirtiness and commit referent stuff.

cassandras-lies commented:
Do you want me to vendor boost-cmake and openssl-cmake then? They're both dead projects, and that will do away with the ugly patching stuff.

cassandras-lies commented:
sparkmobile is patched too, but we control that repo, so we can just push the fix to the real repo. I'm not sure why that wasn't done in the first place really.

julian-CStack commented:
> Do you want me to vendor boost-cmake and openssl-cmake then? They're both dead projects, and that will do away with the ugly patching stuff.

What about adding those as subtrees? We can then commit the changes to the main repo and do away with patching during build? 




> sparkmobile is patched too, but we control that repo, so we can just push the fix to the real repo. I'm not sure why that wasn't done in the first place really.

That is just the addition of the two cmakelists IIRC. I have no objection to actually getting those in to sparkmobile

cassandras-lies committed:
Delete boost-cmake submodule in preparation of adding it as a subtree.

cassandras-lies committed:
Add 'src/deps/boost-cmake/' from commit '40cb41d86eab0d7fdc18af4b04b733f8cc852d2a'

git-subtree-dir: src/deps/boost-cmake
git-subtree-mainline: 1a258d72f049a73f14bdb3d9f748e4dfb9be6d7b
git-subtree-split: 40cb41d86eab0d7fdc18af4b04b733f8cc852d2a

cassandras-lies committed:
Delete openssl-cmake submodule in preparation of adding it as a subtree.

cassandras-lies committed:
Add 'src/deps/openssl-cmake/' from commit 'da217f13d734fe51ccc2d9b986d5f16ccba14bdd'

git-subtree-dir: src/deps/openssl-cmake
git-subtree-mainline: 65e2605a804eafa965ef39db57d755761cc10027
git-subtree-split: da217f13d734fe51ccc2d9b986d5f16ccba14bdd

cassandras-lies committed:
Allow compilation of boost-cmake using CMake 4 and Clang 17.

cassandras-lies committed:
Allow compilation of openssl-cmake using CMake 4 and Clang 17.

cassandras-lies committed:
Delete sparkmobile submodule in preparation of adding it as a subtree.

cassandras-lies committed:
Add 'src/deps/sparkmobile/' from commit 'f38d67de131d263c9a8770a5449850a5af565616'

git-subtree-dir: src/deps/sparkmobile
git-subtree-mainline: dbb31d472887132bc9f9b56fc737f3073fe9509c
git-subtree-split: f38d67de131d263c9a8770a5449850a5af565616

cassandras-lies committed:
Add CMakeLists.txt to sparkmobile.

cassandras-lies committed:
Fix secp256k1 compilation on Windows.

cassandras-lies committed:
Add pthread boost patch.

cassandras-lies committed:
Remove now useless prebuild scripts.

cassandras-lies committed:
Remove old macOS/iOS patch scripts.

cassandras-lies committed:
Merge branch 'main' into fix-build

cassandras-lies commented:
Ok, so I added all the dependencies into subtrees and moved the patches into the actual directories. (The boost patches are still replacement files in the `src/deps/boost-cmake/patches` directory, because the boost source is pulled from a network repository.)

julian-CStack commented:
The podspec files for ios and macos work ok without calling extras now? I assume they must. I can test later and tweak if needed.

Was the deletion of the run_build.sh along side the macos podspec while keeping that along side the ios podspec intentional?

Everything else looks good to me.

julian-CStack reviewed (approved):
LGTM

If any issues arise we can fix later
